lua/rocks-config/init.lua

```diff
@@ -99,8 +99,39 @@ local function check_for_duplicate(plugin_name, config_basename, mod_name)
     end
 end
 
+---Load a config and register any errors that happened while trying to load it.
+---Returns false if the module was not found and true if it was, even if errors happened.
+---@param plugin_name string The plugin that is being configured
+---@param config_basename string The basename of the configuration module.
+---@param mod_name string The configuration module to load.
+---@return boolean
+local function load_config(plugin_name, config_basename, mod_name)
+    local ok, result = pcall(function()
+        return try_load_config(mod_name)
+    end)
+
+    if not ok then
+        -- Module was found but failed to load.
+        table.insert(rocks_config.failed_to_load, { plugin_name, config_basename, result })
+        return true
+    end
+
+    if type(result) ~= "boolean" then
+        error("try_load_config did not return boolean as expected.")
+    end
+
+    return result
+end
+
+---Check if any errors were registered during setup.
+---@return boolean
+local function errors_found()
+    return #rocks_config.duplicate_configs_found > 0 and #rocks_config.failed_to_load > 0
```

Comment:
question: shouldn't this be

```suggestion
    return #rocks_config.duplicate_configs_found > 0 or #rocks_config.failed_to_load > 0
```

or

```suggestion
    return #rocks_config.duplicate_configs_found + #rocks_config.failed_to_load > 0
```

Reply:
It should and I shouldn't do last minute refactors without re-testing before comitting :facepalm:

Reply:
I'll set up some test infrastructure soon 😃